docs: update for nuxt3

mctweb opened the pull request:
Nuxt3 has errors that originate from the sdk unless we manually add eventsource to the plugin

ganigeorgiev commented:
No, this is not specific to Nuxt and the `EventSource` polyfill is necessary only if you are using the realtime APIs in a server-side/ReactNative context. See https://github.com/pocketbase/js-sdk?tab=readme-ov-file#nodejs-via-npm.

If the error you are experiencing is originating from the usage of `authWithOAuth2` call (aka. the "all-in-one" flow) then using it in a server-side context is wrong because this method needs to open a browser window and maintain a realtime connection that will receive the result of the OAuth2 handling. If you want to handle OAuth2 entirely on the server-side then you'll have to use the "Manual code exchange flow" (the second tab in https://pocketbase.io/docs/authentication/#authenticate-with-oauth2).

With that said, I don't recommend using Nuxt in a SSR framework especially if you are not sure where your code is running (browser, server, mixed?). If you haven't already, please read https://pocketbase.io/docs/how-to-use/.

mctweb commented:
I was using the realtime api in an onMounted call, in a client only component. 
It worked locally, however when deploying the frontend to netlify, issues presented.  I removed all realtime API calls from Nuxt, and the issue was still present until I added the fix specified. Perhaps it is a nuxt/netlify issue. But I thought the workaround may be useful if someone else comes across the same issue.

Here is the log from Netlify:
`Mar 9, 09:30:46 PM: d18f0ea9 ERROR  Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"ReferenceError: EventSource is not defined","reason":{"errorType":"ReferenceError","errorMessage":"EventSource is not defined","stack":["ReferenceError: EventSource is not defined","    at RealtimeService.initConnect (file:///var/task/node_modules/pocketbase/dist/pocketbase.es.mjs:1:13209)","    at file:///var/task/node_modules/pocketbase/dist/pocketbase.es.mjs:1:12962","    at new Promise (<anonymous>)","    at RealtimeService.connect (file:///var/task/node_modules/pocketbase/dist/pocketbase.es.mjs:1:12857)","    at RealtimeService.subscribe (file:///var/task/node_modules/pocketbase/dist/pocketbase.es.mjs:1:10252)","    at RecordService.subscribe (file:///var/task/node_modules/pocketbase/dist/pocketbase.es.mjs:1:17585)","    at useUser (file:///var/task/chunks/build/server.mjs:22199:34)","    at setup (file:///var/task/chunks/build/server.mjs:27297:26)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: ReferenceError: EventSource is not defined","    at process.<anonymous> (file:///var/runtime/index.mjs:1276:17)","    at process.emit (node:events:518:28)","    at emitUnhandledRejection (node:internal/process/promises:250:13)","    at throwUnhandledRejectionsMode (node:internal/process/promises:385:19)","    at processPromiseRejections (node:internal/process/promises:470:17)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}`

ganigeorgiev commented:
Again, in order to have "Unhandled Promise Rejection" you mostly likely have somewhere a call that is using the realtime APIs (`authWithOAuth2()` also uses the realtime APIs under the hood).

Feel free to create a _minimal_ reproducible and I'd try to investigate it in more details but suggesting registering it by default without having a clear understanding why it is needed would be wrong.
